usecases/schema/tenant.go

```diff
@@ -38,32 +38,44 @@ func (h *Handler) AddTenants(ctx context.Context,
 	principal *models.Principal,
 	class string,
 	tenants []*models.Tenant,
-) (uint64, error) {
+) ([]*models.Tenant, error) {
 	if err := h.Authorizer.Authorize(principal, "update", tenantsPath); err != nil {
-		return 0, err
+		return nil, err
 	}
 
 	validated, err := validateTenants(tenants)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
 
 	if err = validateActivityStatuses(validated, true); err != nil {
-		return 0, err
+		return nil, err
 	}
 
 	request := api.AddTenantsRequest{
 		ClusterNodes: h.clusterState.Candidates(),
 		Tenants:      make([]*api.Tenant, 0, len(validated)),
 	}
+	tNames := make([]string, len(validated))
 	for i, tenant := range validated {
+		tNames[i] = tenant.Name
 		request.Tenants = append(request.Tenants, &api.Tenant{
 			Name:   tenant.Name,
 			Status: schema.ActivityStatus(validated[i].ActivityStatus),
 		})
 	}
 
-	return h.schemaManager.AddTenants(class, &request)
+	if _, err = h.schemaManager.AddTenants(class, &request); err != nil {
+		return nil, err
+	}
+
+	// we get the new state to return correct status
+	// specially in FREEZING and UNFREEZING
+	uTenants, _, err := h.schemaManager.QueryTenants(class, tNames)
+	if err != nil {
+		return nil, err
+	}
+	return uTenants, err
```

Comment:
**thought:** for `FREEZING` and `UNFREEZING` both shall not be set by the user,  however we can get the status directly from add tenants instead of query to safe requests , something like `UpdateTenants` [here](https://github.com/weaviate/weaviate/pull/5119/files#diff-73e4c57bc541d7a6da82489dd7726a70668bceece88af8f9d75a4017c2722142R233)


but the question is users can't create a `FROZEN` tenant directly! 
and they can't come to a point where they create a tenant and then it `UNFREEZE` because it will be already exists 🤔

Reply:
Makes sense! Will undo these changes so that the `params.Body` tenants are returned to the user upon successful creation as before